sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala

```diff
@@ -261,41 +261,45 @@ class QueryExecutionErrorsSuite
     }
   }
 
-  test("UNSUPPORTED_OPERATION: timeZoneId not specified while converting TimestampType to Arrow") {
+  test("UNSUPPORTED_FEATURE: timeZoneId not specified while converting TimestampType to Arrow") {
     checkErrorClass(
       exception = intercept[SparkUnsupportedOperationException] {
         ArrowUtils.toArrowSchema(new StructType().add("value", TimestampType), null)
```

Comment:
I wonder is it possible to trigger the error from user space? If not, let use `INTERNAL_ERROR` or `IllegalStateException`.

Reply:
Let's replace the check https://github.com/apache/spark/blob/88e53e59e746a6d3aadac3cdc5fcbd74d90ee0e9/sql/catalyst/src/main/scala/org/apache/spark/sql/util/ArrowUtils.scala#L51 by an assert, and remove the test. `ArrowUtils.toArrowType()` is internal API, BTW.